pkg/controller/core/workload_controller.go

```diff
@@ -234,7 +255,7 @@ func syncAdmissionCheckConditions(conds []metav1.Condition, queueChecks []string
 
 func (r *WorkloadReconciler) reconcileNotReadyTimeout(ctx context.Context, req ctrl.Request, wl *kueue.Workload) (ctrl.Result, error) {
 	log := ctrl.LoggerFrom(ctx)
-	countingTowardsTimeout, recheckAfter := r.admittedNotReadyWorkload(wl, realClock)
+	countingTowardsTimeout, recheckAfter := r.quotaReservedNotReadyWorkload(wl, realClock)
```

Comment:
NotReadyTimeout should actually count since Admitted=True (the new Admitted condition)